src/xunit.performance.api/Profilers/Etw/SimpleTraceEventParser.cs

```diff
@@ -0,0 +1,378 @@
+        ///     2. Process started/stopped within the ETW session.
+        /// </remarks>
+        public IReadOnlyCollection<Process> GetProfileData(ScenarioExecutionResult scenarioInfo)
+        {
+            var processes = new List<Process>();
+            Module defaultNtoskrnlModule = null;
+            var pmcSamplingIntervals = new Dictionary<int, long>();
+
+            Func<ProcessTraceData, bool> IsOurProcess = (ProcessTraceData obj) => {
+                return obj.ProcessID == scenarioInfo.ProcessExitInfo.ProcessId
+                    || processes.Any(process => process.Id == obj.ProcessID || process.Id == obj.ParentID);
+            };
+
+            using (var source = new ETWTraceEventSource(scenarioInfo.EventLogFileName))
+            {
+                if (source.EventsLost > 0)
+                    throw new Exception($"Events lost in trace '{scenarioInfo.EventLogFileName}'");
+
+                ////////////////////////////////////////////////////////////////
+                // Process data
+                var parser = new KernelTraceEventParser(source);
+                parser.ProcessStart += (ProcessTraceData obj) => {
+                    if (IsOurProcess(obj))
+                    {
+                        var process = new Process(obj.ImageFileName, obj.ProcessID, obj.ParentID);
+                        process.LifeSpan.Start = obj.TimeStamp;
+                        processes.Add(process);
+                    }
+                };
+                parser.ProcessStop += (ProcessTraceData obj) => {
+                    if (IsOurProcess(obj))
+                        processes.Single(process => process.Id == obj.ProcessID).LifeSpan.End = obj.TimeStamp;
+                };
+
+                ////////////////////////////////////////////////////////////////
+                // Image/Module data
+                parser.ImageLoad += (ImageLoadTraceData obj) => {
+                    var process = processes.SingleOrDefault(p => p.Id == obj.ProcessID);
+                    if (process == null)
+                        return;
+
+                    var module = new Module(obj.FileName, obj.ImageChecksum) {
+                        AddressSpace = new AddressSpace(obj.ImageBase, (uint)obj.ImageSize)
+                    };
+                    module.LifeSpan.Start = obj.TimeStamp;
+
+                    process.Modules.Add(module);
+                };
+                parser.ImageUnload += (ImageLoadTraceData obj) => {
+                    var process = processes.SingleOrDefault(p => p.Id == obj.ProcessID);
+                    if (process == null)
+                        return;
+
+                    // Check if the unloaded module is on the list.
+                    // The module must be loaded, in the same address space, and same file name.
+                    var module = process.Modules
+                        .SingleOrDefault(m => obj.TimeStamp > m.LifeSpan.End && obj.Equal(m));
```

Comment:
I don't think that this condition can ever evaluate to true and so SingleOrDefault will never return you a valid module.  m.LifeSpan.End starts as DateTime.MaxValue which is always greater than obj.TimeStamp.

Reply:
You are right, this is wrong. I'll fix it.

---
In reply to: [146055343](https://github.com/Microsoft/xunit-performance/pull/242#discussion_r146055343) [](ancestors = 146055343)